### PR TITLE
texlive: update to texlive 2019

### PR DESCRIPTION
**Pull request by gardner48**

### CODE REVIEW

File: var/spack/repos/builtin/packages/texlive/package.py [hunk 1]
```diff
@@ -26,8 +26,8 @@ class Texlive(Package):
     # itself is stable.  Don't let that fool you though, it's still
     # installing TeX **LIVE** from e.g. ctan.math.... below, which is
     # not reproducible.
-    version('live', '946701aa28ca1f93e55e8310ce63fbf8',
-            url='ftp://tug.org/historic/systems/texlive/2018/install-tl-unx.tar.gz')
+    version('live', sha256='44aa41b5783e345b7021387f19ac9637ff1ce5406a59754230c666642dfe7750',
+            url='ftp://tug.org/historic/systems/texlive/2019/install-tl-unx.tar.gz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/texlive/package.py]: I wonder if we should keep the old version around and rename `live` to `2018` and `2019`. Thoughts @hartzell?

Comment (gardner48) [reply var/spack/repos/builtin/packages/texlive/package.py]: Trying to build with the old version gives the following error:
```
Loading http://ctan.math.washington.edu/tex-archive/systems/texlive/tlnet/tlpkg/texlive.tlpdb
=============================================================================
./install-tl: The TeX Live versions of the local installation
and the repository being accessed are not compatible:
      local: 2018
 repository: 2019
Perhaps you need to use a different CTAN mirror?
(For more, see the output of install-tl --help, especially the
 -repository option.  Online via http://tug.org/texlive/doc.)
=============================================================================
```

Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/texlive/package.py]: Yeah I have updated this in the past, and you only get the latest version. They delete old versions.

Comment (healther) [reply var/spack/repos/builtin/packages/texlive/package.py]: @adamjstewart IIRC they offer older versions for package repository managers (think distributions) but I never bothered to contact them to find out what we would need to do in order to become one of their sanctioned managers as we started to just use the system one when we started using containers to deploy our software stack

**Pull request merged**